### PR TITLE
Add Dsyev and tests

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
Okay, let's try this again. My apologies. The test code is identical to the old one, and the first half of native/dsyev.go is. I reconstructed the back half of the code.

**Comment by btracey** [comment]
PTAL @kortschak @vladimir-ch

File: native/dsyev.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	rmax := math.Sqrt(bignum)
+
+	// Scale matrix to allowable range, if necessary.
+	anrm := impl.Dlansy(lapack.MaxAbs, uplo, n, a, lda, work)
+	scaled := false
+	var sigma float64
+	if anrm > 0 && anrm < rmin {
+		scaled = true
+		sigma = rmin / anrm
+	} else {
```
Comment (vladimir-ch) [inline native/dsyev.go]: `} else if anrm > rmax {`

File: native/dsyev.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+//
+// w contains the eigenvalues in ascending order upon return. w must have length
+// at least n, and Dsyev will panic otherwise.
+//
+// On entry, a contains the elements of the symmetric matrix A in the triangular
+// portion specified by uplo. If jobz == lapack.EigDecomp a contains the
+// orthonormal eigenvectors of A on exit, otherwise on exit the specified
+// triangular region is overwritten.
+//
+// Work is temporary storage, and lwork specifies the usable memory length. At minimum,
```
Comment (vladimir-ch) [inline native/dsyev.go]: Reformat this paragraph?

File: lapack64/lapack64.go [hunk 3]
```diff
@@ -324,6 +324,26 @@ func Pocon(a blas64.Symmetric, anorm float64, work []float64, iwork []int) float
...
+//
+// w contains the eigenvalues in ascending order upon return. w must have length
+// at least n, and Dsyev will panic otherwise.
+//
+// On entry, a contains the elements of the symmetric matrix A in the triangular
+// portion specified by uplo. If jobz == lapack.EigDecomp a contains the
+// orthonormal eigenvectors of A on exit, otherwise on exit the specified
+// triangular region is overwritten.
+//
+// Work is temporary storage, and lwork specifies the usable memory length. At minimum,
```
Comment (vladimir-ch) [inline lapack64/lapack64.go]: This one too?

**Comment by vladimir-ch** [comment]
LGTM

File: lapack64/lapack64.go (hunk 3 above)
Comment (btracey) [inline lapack64/lapack64.go]: Done.

File: native/dsyev.go (hunk 2 above)
Comment (btracey) [inline native/dsyev.go]: Done.

File: native/dsyev.go (hunk 1 above)
Comment (btracey) [inline native/dsyev.go]: Done.

**Comment by btracey** [comment]
Woo!

**Pull request merged**